### ISSUE TITLE
False positive for `prefer_early_return` when using `throw` for short-circuiting

### ISSUE THREAD

**Issue by yurii-prykhodko-solid** [solid-software/solid_lints#182]
Repro:

```dart
void repro({bool cond = false}) {
  if (cond) {
    throw '';
  }

  return;
}
```

```
lib/tmp.dart:2:3 • Use reverse if to reduce nesting • prefer_early_return • WARNING
```

Version: 0.2.3
